build.gradle

```diff
@@ -1,8 +1,270 @@
+    licenseFile = rootProject.file('LICENSE.txt')
+    noticeFile = rootProject.file('NOTICE.txt')
+}
+
+configurations.all {
+    if (it.state != Configuration.State.UNRESOLVED) return
+    resolutionStrategy {
+        force 'commons-codec:commons-codec:1.14'
+        force 'org.slf4j:slf4j-api:1.7.26'
+        force 'org.apache.httpcomponents:httpclient:4.5.10'
+        force 'com.google.code.findbugs:jsr305:3.0.2'
+        force 'com.fasterxml.jackson.core:jackson-databind:2.11.2'
+        force 'com.fasterxml.jackson.core:jackson-core:2.11.2'
+        force 'commons-cli:commons-cli:1.3.1'
+        force 'commons-logging:commons-logging:1.2'
+        force 'com.google.guava:guava:25.1-jre'
+        force 'org.bouncycastle:bcprov-jdk15on:1.67'
+        force 'org.apache.commons:commons-lang3:3.4'
+        force 'org.apache.santuario:xmlsec:2.2.0'
+        force 'com.fasterxml.woodstox:woodstox-core:6.2.1'
+        force 'io.netty:netty-buffer:4.1.49.Final'
+        force 'io.netty:netty-common:4.1.49.Final'
+        force 'io.netty:netty-handler:4.1.49.Final'
+        force 'io.netty:netty-transport:4.1.49.Final'
+        force 'com.fasterxml.jackson.core:jackson-annotations:2.11.2'
+        force 'net.sf.jopt-simple:jopt-simple:5.0.4'
+        force 'org.scala-lang:scala-library:2.12.10'
+        force 'org.scala-lang:scala-reflect:2.12.10'
+        force 'org.cryptacular:cryptacular:1.1.4'
+        force 'org.apache.httpcomponents:httpcore:4.4.12'
+        exclude group: "org.elasticsearch", module: "securemock"
+    }
+}
+
+configurations.testCompile {
+    if (it.state != Configuration.State.UNRESOLVED) return
+    resolutionStrategy {
+        force 'org.apache.kafka:kafka-clients:2.0.1:test'
+    }
+}
+
+dependencies {
+    compile('org.elasticsearch.plugin:transport-netty4-client:7.10.2') {
+        exclude group: 'org.elasticsearch', module: 'jna'
+        exclude group: 'com.vividsolutions', module: 'jts'
+        exclude group: 'org.apache.logging.log4j', module: 'log4j-api'
+    }
+    compile 'com.google.guava:guava:25.1-jre'
+    compile 'com.google.errorprone:error_prone_annotations:2.1.3'
+    compile 'org.greenrobot:eventbus:3.2.0'
+    compile 'commons-cli:commons-cli:1.3.1'
+    compile 'org.bouncycastle:bcprov-jdk15on:1.67'
+    compile('com.fasterxml.jackson.core:jackson-databind:2.11.2') {
+        exclude group: 'com.fasterxml.jackson.core', module: 'ackson-core'
+    }
+    compile('org.apache.logging.log4j:log4j-slf4j-impl:2.11.1') {
+        exclude group: 'org.apache.logging.log4j', module: 'log4j-api'
+    }
+    compile('org.ldaptive:ldaptive:1.2.3') {
+        exclude group: 'commons-cli', module: 'commons-cli'
+        exclude group: 'org.slf4j', module: 'slf4j-api'
+    }
+    compile('org.apache.httpcomponents:httpclient-cache:4.5.3') {
+        exclude group: 'org.apache.httpcomponents', module: 'httpcore'
+    }
+    compile('org.elasticsearch.client:elasticsearch-rest-high-level-client:7.10.2') {
```

Comment:
Please define variable and reuse for same versions. Eg: 7.10.2 is used for a couple of dependencies.

Reply:
Done.